src/main/java/com/github/splendor_mobile_game/game/model/Game.java

```diff
@@ -128,10 +128,13 @@ private ArrayList<Card> getRandomCards(ArrayList<Card> cardList, int amount) {
             list.add(i);
         }
 
+
         Random rand = new Random();
         while(list.size() > 0) {
             int index = rand.nextInt(list.size()); // Get random index
             array.add(cardList.get(index));
+            //If the card was drawn it would not be used at any time by the Game class (I think xd)
+            cardList.remove(index);
```

Comment:
What you're doing here is removing a card from list that we are drawning from. It's `already implemented` basing on `list` array. 
What you need to do is to store `local class variable` which will store all available cards, on card drawn remove drawn element from this arraylist AND refactor getRandomCards() to support that solution.